### PR TITLE
if_ether: add ether_arp definition

### PR DESCRIPTION
**Pull request by zhhyu7**
## Summary
We will integrate network management tools in future projects. In the process of porting the tools, we encounter some situations where the structure is not defined, or the returned data types do not match the expectations. Refer to the common implementation of other systems and add relevant definitions.

## Testing
sim:local

### CODE REVIEW

**Review by jerpelea — APPROVED** [review]

**Comment by jerpelea** [comment]
please add commit message describing the change

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/netinet/if_ether.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef _INCLUDE_NETINET_IF_ETHER_H
+#define _INCLUDE_NETINET_IF_ETHER_H
```
Comment (pkarashchenko) [inline include/netinet/if_ether.h:22]:
```suggestion
#ifndef __INCLUDE_NETINET_IF_ETHER_H
#define __INCLUDE_NETINET_IF_ETHER_H
```

File: include/netinet/if_ether.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  uint8_t arp_tpa[4];             /* target protocol address */
+};
+
+#define arp_hrd ea_hdr.ar_hrd
+#define arp_pro ea_hdr.ar_pro
+#define arp_hln ea_hdr.ar_hln
+#define arp_pln ea_hdr.ar_pln
+#define arp_op  ea_hdr.ar_op
+
+#endif /* _INCLUDE_NETINET_IF_ETHER_H */
```
Comment (pkarashchenko) [inline include/netinet/if_ether.h:64]:
```suggestion
#endif /* __INCLUDE_NETINET_IF_ETHER_H */
```

File: include/netinet/if_ether.h [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+/**
+ * Ethernet Address Resolution Protocol.
+ *
+ * See RFC 826 for protocol description.  Structure below is adapted
+ * to resolving internet addresses.  Field names used correspond to
+ * RFC 826.
+ */
+
+struct  ether_arp
```
Comment (pkarashchenko) [inline include/netinet/if_ether.h:49]: remove 2 spaces

Comment (zhhyu7) [reply include/netinet/if_ether.h:49]: @pkarashchenko I corrected these issue in another PR. https://github.com/apache/nuttx/pull/9188

**Pull request merged**